### DISCUSSION TITLE
Nested Modules

### DISCUSSION THREAD

**Discussion by sanchez** [nuxt/nuxt#19628] [Questions]
Hi I was wondering if it is possible to have a nuxt module import and configure a different nuxt module within itself and have that information exposed through the `.nuxt` folder via `nuxt prepare`?

Bit more explanation. I have a UI Nuxt module I am creating and within that module I would like to be able to configure the tailwindcss nuxt module with some specific configuration, then I want the UI module to expose some UI Components under a specific prefix. Then I have another module where I want to build off of the UI Module (e.g. detailed controls or something similar), I want to be able to have all the UI Module components available in this other module with intellisense and type checking, which I think means the `.nuxt` folder of the module has to have all the UI Module definitions in it?

At the moment I am using pnpm and have followed a setup similar to this: https://vueschool.io/articles/vuejs-tutorials/scalable-nuxt-3-monorepos-with-pnpm-workspaces/

Any help and pointers appreciated, apologies if this has been asked else where I couldn't find other material.

**Comment by Baroshem** [comment]
Hey,

You can technically install the module as a part of another module. I am doing it in the nuxt-security module to install and use nuxt-csurf.

https://github.com/Baroshem/nuxt-security/blob/fb5fe242ad32e9ef095a3d67d7eda05c962f4a46/src/module.ts#LL170C3-L170C3

But not sure about the rest. This is a bit advanced use case that I have not tried yet

**Reply by sanchez** [reply]
Interesting, so you can install modules through the use of `installModule` but it doesn't appear to put the installed types into the `.nuxt` folder. Instead you are using a nuxt app labelled `playground` to generate the `.nuxt` folder and linking typescript for the whole repo into that folder? Thank you providing the examples, I think I would still like to have isolated `.nuxt` folders per module because I plan on having a monorepo but I want people to be able to la carte style pick their packages they want.
